pkg/git/git.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Codefresh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/log"
+	"github.com/codefresh-io/cli-v2/pkg/util"
+)
+
+
+var (
+	requiredGitHubScopes = []string{ "repo", "admin:repo_hook" }
+)
+
+
+func VerifyToken(ctx context.Context, provider string, token string) error {
+	providerToVerifier := map[string]func(context.Context, string)error {
+		"github": verifyGitHubTokenScope,
+	}
+
+	verifier := providerToVerifier[provider]
+	if verifier == nil {
+		return verifyTokenScopeFallback(ctx, provider)
+	}
+
+	return verifier(ctx, token)
+}
+
+func verifyGitHubTokenScope(ctx context.Context, token string) error {
+	errMessage := "The provided git token is missing one or more of the required scopes:" + util.StringifyArray(requiredGitHubScopes)
```

Comment:
you don't need `StringifyArray`, just do: `fmt.Sprintf("some message: %v", array)`

Reply:
also make sure your error messages are all lowercase

Reply:
with %v it looks ugly :(